eng/tools/RepoTasks/CreateFrameworkListFile.cs

```diff
@@ -57,12 +57,34 @@ public override bool Execute()
                     (f.Filename.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) || f.IsNative))
                 .OrderBy(f => f.Filename, StringComparer.Ordinal))
             {
-                var element = new XElement(
-                    "File",
-                    new XAttribute("Type", f.IsNative ? "Native" : "Managed"),
-                    new XAttribute(
-                        "Path",
-                        Path.Combine(f.PackagePath, f.Filename).Replace('\\', '/')));
+                string path = Path.Combine(f.PackagePath, f.Filename).Replace('\\', '/');
+                string type = f.IsNative ? "Native" : "Managed";
+                var element = new XElement("File", new XAttribute("Path", path));
+
+                if (path.StartsWith("analyzers/"))
+                {
+                    type = "Analyzer";
+
+                    if (path.EndsWith(".resources.dll"))
+                    {
+                        // omit analyzer resources
+                        continue;
+                    }
+
+                    var pathParts = path.Split('/');
+
+                    if (pathParts.Length < 3 || !pathParts[1].Equals("dotnet", StringComparison.Ordinal) || pathParts.Length > 4)
+                    {
+                        Log.LogError($"Unexpected analyzer path format {path}.  Expected  'analyzers/dotnet(/language)/analyzer.dll");
+                    }
+
+                    if (pathParts.Length > 3)
+                    {
+                        element.Add(new XAttribute("Language", pathParts[2]));
```

Comment:
> I specifically want it to be greater than 3. That means there is a language directory. If it's only 3 there is no language directory.

In that case, this line should be
```suggestion
                        element.Add(new XAttribute("Language", pathParts[3]));
```

Reply:
No, that would be using the file name as language.  Length is 1-base, array indexes are 0-base.  When we have more than 3 parts we want to take the 3rd part (index 2).

Reply:
I added a comment, please see if that clarifies it.

Reply:
Yeah. I just somehow forgot the filename portion of the path 🤦

Reply:
It's ok, you were spot on for catching an off-by-one error which this does appear to be.  Better to add a comment.